src/sasctl/core.py

```diff
@@ -712,7 +741,14 @@ def get_auth(self, username=None, password=None, token=None):
 
         # If explicit username & password were provided, use them.
         if username is not None and password is not None:
-            return self.get_oauth_token(username, password)
+            return self.get_oauth_token(
+                username, password
+            )
+        elif client_id is not None and client_secret is not None:
```

Comment:
This additional elif with a return embedded is throwing a [PYL-R1705 linter error](https://deepsource.io/gh/sassoftware/python-sasctl/run/249aba1b-cecd-45fa-8ba9-e8acd8dfa4d5/python/PYL-R1705?listindex=0). For the time being can we split this out to two separate if statements.

In the future, it looks like this function could use a refactor for two elements:
1. Combine if statements into fully encapsulated set of cases (similar to the switch/case implementation in Python 3.10).
```
if username and password:
    return ...
elif client_id and client_secret:
    return ...
elif token:
    return ...
else:
    try:
        return kerberos
    ...
    prompt user input
    return user input
```
2. Remove double negative conditionals.
`if username is not None and password is not None:` can be written as `if username and password:`

Reply:
Don, point 1.